docs/user-guide.md

```diff
@@ -328,6 +328,15 @@ performed.
 *   `HANDLE_EA_EL3_FIRST`: When defined External Aborts and SError Interrupts
     will be always trapped in EL3 i.e. in BL31 at runtime.
 
+*   `HW_ASSISTED_COHERENCY`: On most ARM systems to-date, platform-specific
+    software operations are required for CPUs to enter and exit coherency.
+    However, there exists newer systems where CPUs' entry to and exit from
+    coherency is managed in hardware. Such systems require software to only
```

Comment:
@jeenu-arm  thanks. very interesting. I might have expected coherency to be active only after MMUs are active. I am guessing the patch is due to sizeable performance advantage by skipping the syncs off. if PTE driven coherency, then initial stages of cache invalidations might still be valid.

but thanks for the explanation. will keep a watch.